perses/annihilation/relative.py

```diff
@@ -2629,3 +2629,1504 @@ def _handle_exceptions(self):
+
+                             #define switching function
+                             "switch = step(-t) + step(t) * step(1-t) * (1 + (t^3) * (-10 + t * (15 - (6 * t))));",
+                             "t = (r_eff - r_switch) / (r_cutoff - r_switch);",
+
+                             #define topological term
+                             f"c_RF = {ONE_4PI_EPS0} * chargeProd;",
+                             "chargeProd = charge1 * charge2;",
+                             "charge1 = ((charge_old1 * {old_bool_string1}) + (charge_new1 * {new_bool_string1})) * {scale_bool_string1};", 
+                             "charge2 = ((charge_old2 * {old_bool_string2}) + (charge_new2 * {new_bool_string2})) * {scale_bool_string2};",
+
+                             #define reaction field u_RF
+                             "u_RF = (1 / r_eff) + ((eps_RF - 1) * (r_eff^2)) / ((r_cutoff^3) * (1 + (2 * eps_RF)));",
+
+                             #r_eff
+                             "r_eff = sqrt(r^2 + w^2);",
+
+                             #4th dimension:
+                             #"w = step(unique_new1 + unique_new2 + unique_old1 + unique_old2 - 0.1) * r_cutoff * ({old_bool_string1} - {new_bool_string1}) * {w_scale};",
+                             "w = is_unique_old * r_cutoff * w_scale * (1 - lambda_0_electrostatics_old) + is_unique_new * w_scale * (1 - lambda_0_electrostatics_new);",
+                             "is_unique_old = step(unique_old1 + unique_old2 - 0.1);",
+                             "is_unique_new = step(unique_new1 + unique_new2 - 0.1);",
+
+                             # switch parameter
+                             "r_switch = r_cutoff * {switching_ratio};",
+
+                             # reaction field parameters
+                             "eps_RF = {eps_RF};",
+                             "r_cutoff = {r_cutoff};",
+                             "w_scale = {w_scale};"
+                             ]
+    # predefined RF parameters (fed to the `expression` via `format`): [scale_bool_string, old_bool_string, new_bool_string, w_scale, switching_ratio, eps_RF, r_cutoff]
+    # RF per_particle_parameters: [scale_bools, old/new_bool_string, unique_old, unique_new, core]
+
+    #electrostatic exception
+    _default_RF_exception_expr_list = copy.deepcopy(_default_RF_expr_list)
+    _default_RF_exception_expr_list[0] = "U_electrostatics * {scale_bool_string};"
+    _default_RF_exception_expr_list[5] = "chargeProd = select(1 - environment_region, chargeProd_old * {old_bool_string} + chargeProd_new * {new_bool_string}, chargeProd_old);"
+    _default_RF_exception_expr_list[6] = ''
+    _default_RF_exception_expr_list[7] = ''
+    _default_RF_exception_expr_list[10] = "w = (is_unique_old * r_cutoff * w_scale * (1 - lambda_0_electrostatics_exceptions_old)) + (is_unique_new * w_scale * (1 - lambda_0_electrostatics_exceptions_new));"
```

Comment:
Again, why does `r_cutoff * w_scale` appear in the first term, but only `w_scale` in the second term?